docs/src/extend/firmware-management/building-image/rugpi/tutorial.md

```diff
@@ -0,0 +1,141 @@
+
+## Setting up your build system
+
+Building images using Rugpi is considerably faster and easier when compared to The Yocto Project, so it makes it a great tool for beginners.
+
+The following tools are required to build an image:
+
+* Docker Engine
+
+:::tip
+Images can be built using Rugpi using a CI Workflow. An example for a Github Workflow is included in the template project.
+:::
+
+## Building your image
+
+The [tedge-rugpi-images](https://github.com/thin-edge/tedge-rugpi-image) is project which includes out of the box configuration to perform robust Over-the-Air Operating System updates.
+
+Feel free to clone the project if you want to make your own customizations, however please always refer back to the project if you run into any problems (as it may have changed in the meantime).
+
+
+### Cloning the project
+
+1. Clone the project and change directory to the locally checked out folder
+
+    ```sh
+    git clone https://github.com/thin-edge/tedge-rugpi-image.git
+    cd tedge-rugpi-image
+    ```
+
+2. Install [just](https://just.systems/man/en/chapter_5.html) which is used to run different tasks provided by the project
+
+    ```sh
+    curl --proto '=https' --tlsv1.2 -sSf https://just.systems/install.sh | sudo bash -s -- --to /usr/bin
+    ```
+
+    Check out the [justfile website](https://just.systems/man/en/chapter_5.html) for more installation options.
+
+### Customizing your own image
+
+1. Fork the project
+
+2. Edit the authorized ssh keys for each profile
+
+    Edit the `root_authorized_keys` under each toml file in the `profiles/` directory, and add the public ssh keys whom you want to grant ssh access to for the device. Below shows one example.
+
+    ```toml title="file: profiles/default.toml"
+    #...
+
+    [parameters.ssh]
+    root_authorized_keys = """
+    ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDfhQGWWw73ponAokdNSRZ5cQc9/CIX1TLQgYlr+BtObKoO4UNFP1YSbgK03GjhjeUid+QPmV+UURqxQTqLQoYWqUFP2CYkILFccVPmTvx9HLwupI+6QQKWfMDx9Djfph9GzInymaA5fT7hKppqittFrC/l3lkKgKTX5ohEOGshIbRgtgOYIaW3ByTx3urnaBbYCIgOyOZzSIyS0dUkwsiLu3XjPspgmn3Fs/+vofT/yhBe1carW0UM3ivV0JFfJzrxbCl/F7I2qwfjZXsypjkwlpNupUMuo3xPMi8YvNvyEu4d+IEAqO1dCcdGcxlkiHxrdITIpVLt5mjJ2LauHE/H bootstrap
+    """
+
+    # ...
+    ```
+
+3. Commit the changes
+
+4. You can now add any custom recipes to your project if you want to include additional packages in your image, or configure items.
+
+    See the [Rugpi System Customization](https://oss.silitics.com/rugpi/docs/guide/system-customization) docs for details on how to do this or follow clues from some of the existing recipes included in the [tedge-rugpi-images](https://github.com/thin-edge/tedge-rugpi-image) project.
+
+### Building an image
+
+:::info
+Currently building is only supported on a Linux environment. It is strongly encouraged to use the Github workflow to build the image.
+:::
+
+1. Build an image for your device (machine)
```

Comment:
On my linux box, I have the same issue as for CI https://github.com/thin-edge/tedge-rugpi-image/blob/main/scripts/compress.sh#L6,

i.e. I need to run `xz` with `sudo` to avoid the error `xz: build/tedge_rugpi_default_pi4_20231220.1302.img.xz: Permission denied`

Adding this as a Tip could be useful.

Reply:
I'd rather just make the script a bit more intelligent and do the compression automatically...though I still don't fully understand where the exact error comes from, or if there is a more convenient way to solve it.

Reply:
In the meantime I've added a tip about running the same command using sudo if a "Permission denied" error is experienced. I'll still look for a better way to handle this.

Resolved by 51ec5440a4e622694b97e9c612bf6ae28ac740fd